lms/views/api/canvas/sync.py

```diff
@@ -39,32 +41,5 @@ def sync(self):
         authority = self._request.registry.settings["h_authority"]
         return [group.groupid(authority) for group in groupings]
 
-    @property
-    def _is_speedgrader(self):
-        return "learner" in self._request.json
-
-    def group_set(self):
-        if self._is_speedgrader:
-            return int(self._request.json["learner"].get("group_set"))
-
-        return int(self._request.json["course"].get("group_set"))
-
-    @property
-    def _is_group_launch(self):
-        application_instance = self._request.find_service(
-            name="application_instance"
-        ).get_current()
-        if not application_instance.settings.get("canvas", "groups_enabled"):
-            return False
-
-        try:
-            self.group_set()
-        except (KeyError, ValueError, TypeError):
-            return False
-        else:
-            return True
-
-    def _get_course(self):
-        return self._request.find_service(name="course").get_by_context_id(
-            self._request.json["course"]["context_id"],
-        )
+    def get_course(self, course_id):
+        return self._request.find_service(name="course").get_by_context_id(course_id)
```

Comment:
As these two views are so similar, basically everything I said for the other goes for this